avocado/core/runners/utils/messages.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import sys
+import time
+
+from ... import output
+
+
+class GenericMessage:
+    message_status = None
+
+    @classmethod
+    def _prepare_message(cls, additional_info=None):
+        """Prepare a message dict with some basic information.
+
+        This will add the keyword 'status' and 'time' to all messages.
+
+        :param: addional_info: Any additional information that you
+                               would like to add to the message.
+        :type additional_info: dict
+        :return: message dict which can be send to avocado server
+        :rtype: dict
+        """
+        status = {}
+        if isinstance(additional_info, dict):
+            status = additional_info
+        status.update({'status': cls.message_status,
+                       'time': time.monotonic()})
+        return status
+
+    @classmethod
+    def get(cls, **kwargs):
+        """Creates message base on it's type with  all necessary information.
+
+        :return: message dict which can be send to avocado server
+        :rtype: dict
+        """
+        kwargs = {key: value for (key, value) in kwargs.items()
+                  if value is not None}
+        return cls._prepare_message(additional_info=kwargs)
+
+
+class StartedMessage(GenericMessage):
+    message_status = 'started'
+
+
+class RunningMessage(GenericMessage):
+    """Creates running message without any additional info."""
+    message_status = 'running'
+
+
+class FinishedMessage(GenericMessage):
+    message_status = 'finished'
+
+    @classmethod
+    def get(cls, result, fail_reason=None, returncode=None):  # pylint: disable=W0221
+        """Creates finished message with  all necessary information.
+
+        :param result: test result
+        :type result values for the statuses defined in
+                     :class: avocado.core.teststatus.STATUSES
+        :param fail_reason: parameter for brief specification, of the failed
+                            result.
+        :type fail_reason: str
+        :param returncode: exit status of runner
+        :return: finished message
+        :rtype: dict
+        """
+        return super(FinishedMessage, cls).get(result=result,
+                                               fail_reason=fail_reason,
+                                               returncode=returncode)
+
```

Comment:
Unless I'm missing something, this could be: `super().get(...` Any reason for that?

Reply:
Yes, I will fix that.